### PR TITLE
Grant Cluster Service access to MGMT Key Vautls

### PR DESCRIPTION
**Pull request by janboll**
### What this PR does

Jira: https://issues.redhat.com/browse/ARO-12201
Link to demo recording:

### CODE REVIEW

**Comment by machi1990** [comment]
@miguelsorianod can you also help review this? It's related to the work around certs and secrets persist / retrieval / delete in CS for cluster provision flow

File: dev-infrastructure/templates/mgmt-cluster.bicep [hunk 1]
```diff
@@ -246,6 +249,42 @@ module mgmtKeyVault '../modules/keyvault/keyvault.bicep' = {
   }
 }
 
+module cxClusterServiceKeyVaultAccess '../modules/keyvault/keyvault-secret-access.bicep' = [
+  for role in [
+    'Key Vault Secrets User'
+    'Key Vault Certificate User'
+    'Key Vault Certificates Officer'
+  ]: {
```
Comment (machi1990) [inline dev-infrastructure/templates/mgmt-cluster.bicep:257]: These three roles looks about right for me.
Anything else we need to remove / add @miguelsorianod ?

File: dev-infrastructure/templates/mgmt-cluster.bicep [hunk 2]
```diff
@@ -246,6 +249,42 @@ module mgmtKeyVault '../modules/keyvault/keyvault.bicep' = {
...
+    ]
+  }
+]
+
+module msiCluserServiceKeyVaultAccess '../modules/keyvault/keyvault-secret-access.bicep' = [
+  for role in [
+    'Key Vault Secrets User'
+    'Key Vault Certificate User'
+    'Key Vault Certificates Officer'
+  ]: {
```
Comment (machi1990) [inline dev-infrastructure/templates/mgmt-cluster.bicep:275]: same as https://github.com/Azure/ARO-HCP/pull/867/files#r1852308484 @miguelsorianod

File: config/config.yaml [hunk 3]
```diff
@@ -32,6 +32,7 @@ defaults:
 
   # MGMT cluster specifics
   mgmt:
+    clusterServicePrincipalId: /subscriptions/1d3378d3-5a3f-4712-85a1-2485495dfc4b/resourcegroups/hcp-underlay-{{ .ctx.regionShort }}-svc/providers/Microsoft.ManagedIdentity/userAssignedIdentities/image-sync
```
Comment (geoberle) [inline config/config.yaml:35]:
```suggestion
    clusterServicePrincipalId: /subscriptions/1d3378d3-5a3f-4712-85a1-2485495dfc4b/resourcegroups/hcp-underlay-{{ .ctx.regionShort }}-svc/providers/Microsoft.ManagedIdentity/userAssignedIdentities/clusters-service
```
like this?

File: dev-infrastructure/templates/mgmt-cluster.bicep (hunk 1 above)
Comment (geoberle) [reply dev-infrastructure/templates/mgmt-cluster.bicep:257]: i guess you will need `Key Vault Secrets Officer` as well to be able to create secrets, for things like the the ACR pull secret token etc

File: dev-infrastructure/templates/mgmt-cluster.bicep [hunk 4]
```diff
@@ -246,6 +249,42 @@ module mgmtKeyVault '../modules/keyvault/keyvault.bicep' = {
...
+      roleName: role
+      managedIdentityPrincipalId: clusterServicePrincipalId
+    }
+    dependsOn: [
+      cxKeyVault
+    ]
+  }
+]
+
+module msiCluserServiceKeyVaultAccess '../modules/keyvault/keyvault-secret-access.bicep' = [
```
Comment (geoberle) [inline dev-infrastructure/templates/mgmt-cluster.bicep:270]: typo

File: dev-infrastructure/templates/mgmt-cluster.bicep (hunk 1 above)
Comment (machi1990) [reply dev-infrastructure/templates/mgmt-cluster.bicep:257]: You are right. We need RW and I thought Key Vault Secrets User gives us that by looking at the comment in https://github.com/Azure/ARO-HCP/blob/3932d27cd29f7207bb277468507b276df1aa2055/dev-infrastructure/modules/keyvault/keyvault-secret-access.bicep#L20 but it seems like the comment wrong it should be `Read secret contents including secret portion of a certificate with private key. Only works for key vaults that use the 'Azure role-based access control' permission model.` Non?

File: dev-infrastructure/modules/keyvault/keyvault-secret-access.bicep [hunk 5]
```diff
@@ -25,6 +27,14 @@ var roleResourceIds = {
     'Microsoft.Authorization/roleDefinitions/',
     'b86a8fe4-44ce-4948-aee5-eccb2c155cd7'
   )
+  'Key Vault Certificate User': subscriptionResourceId(
```
[the remark this answers is not in the document]
Comment (machi1990) [reply dev-infrastructure/modules/keyvault/keyvault-secret-access.bicep:30]:
```
Read entire certificate contents including secret and key portion. 
```

[the remark this answers is not in the document]
Comment (janboll) [reply dev-infrastructure/modules/keyvault/keyvault-secret-access.bicep:30]: thanks

File: dev-infrastructure/modules/keyvault/keyvault-secret-access.bicep [hunk 6]
```diff
@@ -25,6 +27,14 @@ var roleResourceIds = {
...
+    'Microsoft.Authorization/roleDefinitions/',
+    'db79e9a7-68ee-4b58-9aeb-b90e7c24fcba'
+  )
+  'Key Vault Certificates Officer': subscriptionResourceId(
```
[the remark this answers is not in the document]
Comment (janboll) [reply dev-infrastructure/modules/keyvault/keyvault-secret-access.bicep:34]: thanks

File: dev-infrastructure/templates/mgmt-cluster.bicep (hunk 4 above)
Comment (janboll) [reply dev-infrastructure/templates/mgmt-cluster.bicep:270]: thanks

File: dev-infrastructure/templates/mgmt-cluster.bicep (hunk 2 above)
Comment (janboll) [reply dev-infrastructure/templates/mgmt-cluster.bicep:275]: changed

File: dev-infrastructure/templates/mgmt-cluster.bicep (hunk 1 above)
Comment (janboll) [reply dev-infrastructure/templates/mgmt-cluster.bicep:257]: changed

**Review by geoberle — APPROVED** [review]

**Pull request merged**